addons/purchase/models/purchase_bill_line_match.py

```diff
@@ -0,0 +1,144 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+
+from odoo.tools import SQL
+from odoo.exceptions import UserError
+
+
+class PurchaseBillMatch(models.Model):
+    _name = "purchase.bill.line.match"
+    _description = "Purchase Line and Vendor Bill line matching view"
+    _auto = False
+    _order = 'product_id, aml_id, pol_id'
+
+    pol_id = fields.Many2one(comodel_name='purchase.order.line')
+    aml_id = fields.Many2one(comodel_name='account.move.line')
+    company_id = fields.Many2one(comodel_name='res.company')
+    partner_id = fields.Many2one(comodel_name='res.partner')
+    product_id = fields.Many2one(comodel_name='product.product')
+    line_qty = fields.Float()
+    line_uom_id = fields.Many2one(comodel_name='uom.uom')
+    qty_invoiced = fields.Float()
+    purchase_order_id = fields.Many2one(comodel_name='purchase.order')
+    account_move_id = fields.Many2one(comodel_name='account.move')
+    line_amount_untaxed = fields.Monetary()
+    currency_id = fields.Many2one(comodel_name='res.currency')
+    state = fields.Char()
+
+    product_uom_id = fields.Many2one(comodel_name='uom.uom', related='product_id.uom_id')
+    product_uom_qty = fields.Float(compute='_compute_product_uom_qty')
+    billed_amount_untaxed = fields.Monetary(compute='_compute_amount_untaxed_fields', currency_field='currency_id')
+    purchase_amount_untaxed = fields.Monetary(compute='_compute_amount_untaxed_fields', currency_field='currency_id')
+    reference = fields.Char(compute='_compute_reference')
+
+    def _compute_amount_untaxed_fields(self):
+        for line in self:
+            line.billed_amount_untaxed = line.line_amount_untaxed if line.account_move_id else False
+            line.purchase_amount_untaxed = line.line_amount_untaxed if line.purchase_order_id else False
+
+    def _compute_reference(self):
+        for line in self:
+            line.reference = line.purchase_order_id.display_name or line.account_move_id.display_name
+
+    def _compute_display_name(self):
+        for line in self:
+            line.display_name = line.pol_id.display_name or line.aml_id.product_id.display_name or line.aml_id.display_name
```

Comment:
This may be confusing because the `line.pol_id.display_name` depends on the Vendor's product name and/or their language
![image](https://github.com/user-attachments/assets/97fcfcfb-e930-4144-8aff-aadbab700d13)
vendor product name

![image](https://github.com/user-attachments/assets/53550998-a528-4c41-b3f4-872105d8b80c)
vendor = French speaker

Which isn't used in the invoice
![image](https://github.com/user-attachments/assets/6fefcb95-80f9-4874-b217-eae8d774bf18)
vendor product name

![image](https://github.com/user-attachments/assets/7f534089-8c3d-4e2b-b2c4-89a1da7444d4)
vendor = French speaker

I don't know how the auto-bill matching works, but I would guess it's hard to do with manually bill matching in this case? Also, this `name` (formerly displayed as "Description") field is editable. So maybe it would be a good idea to include a product column too?

Reply:
Indeed the vendor product name is actually for the vendor... while this view is for internal users
I think including a product column will just add unnecessary complexity to the view...
(though I don't mind including it with `optional="hide"`)
I'm thinking of simply using the `display_name` of the product instead?

Reply:
or maybe the `product_label_section_and_note_field`

Reply:
It is possible for there to be multiple vendor product names for the same vendor though, e.g. something like
![image](https://github.com/user-attachments/assets/0c8ee954-ff50-4246-95b7-3ae1c4334a0c)

So it may be relevant to have both? (I don't really know what invoices typically look like when they arrive). `optional="hide"` or not is fine by me, but I don't really care (that's more a PO's decision :wink: )

Reply:
`product_id.display_name` without any context should simply be the name of the product in our company in the user's language, right?
(makes me wonder if we should use vendor names/languages when adding a product to the PO? theoretically there would be no need to send the PO as we've already been billed)

Reply:
since PO's are not available, let's have a quick chat about it